nan_maybe_pre_43_inl.h

```diff
@@ -153,7 +153,10 @@ inline Maybe<bool> DefineOwnProperty(
   , v8::Handle<v8::String> key
   , v8::Handle<v8::Value> value
   , v8::PropertyAttribute attribs = v8::None) {
-  return obj->GetPropertyAttributes(key) == v8::DontDelete
+  v8::PropertyAttribute current = obj->GetPropertyAttributes(key);
+  return !(current & v8::DontDelete) ||
+                 !(current & v8::ReadOnly) &&
+                     !((attribs ^ current) & ~v8::ReadOnly)
              ? Just<bool>(obj->ForceSet(key, value, attribs))
              : Nothing<bool>();
```

Comment:
I'd argue that this is easier to read written in guard style:
```c++
if (current & v8::DontDelete) return Nothing<bool>();
if (current & v8::ReadOnly) return Nothing<bool>();
// etc.
```

Reply:
I disagree, but since you are doing the reviewing, I changed it. The proposed style does not work, and now it looks messier to me. The way I read the spec it should return `Nothing` iff the property is non-configurable, read-only and flags other than read-only disagree.